### PR TITLE
Wsdl array record

### PR DESCRIPTION
**Pull request by TurboGit**
Just to be approved. I'll merge on my side (I want the merge commit to be done with --log).

### CODE REVIEW

**Comment by anisimkov** [comment]
Why Generate arrayType attribute only for RPC binding ?

**Comment by TurboGit** [comment]
Indeed, a thinko it should be done only when using then encoded encoding and not when in literal encoding style.

**Comment by TurboGit** [comment]
New code in review. Thanks.

File: src/soap/soap-types.adb [hunk 1]
```diff
@@ -1585,11 +1585,14 @@ package body SOAP.Types is
...
+
+      if Encoding = WSDL.Schema.Encoded then
+         Append (Result, " " & SOAP_Enc & ":arrayType=""");
+         Append (Result, Array_Type);
+         Append (Result, '[');
+         Append (Result, AWS.Utils.Image (Natural (O.O'Length)));
+         Append (Result, "]""");
+      end if;
 
       if Encoding = WSDL.Schema.Encoded then
```
Comment (anisimkov) [inline src/soap/soap-types.adb]: 2 same if condition one by one

**Comment by TurboGit** [comment]
Indeed, good catch!

**Comment by TurboGit** [comment]
New code in review.

**Pull request merged**